pl_bolts/models/autoencoders/basic_ae/basic_ae_module.py

```diff
@@ -78,76 +73,82 @@ def training_step(self, batch, batch_idx):
+        tensorboard_logs = {"mse_loss": avg_loss}
 
-        return {
-            'test_loss': avg_loss,
-            'log': tensorboard_logs
-        }
+        return {"test_loss": avg_loss, "log": tensorboard_logs}
 
     def configure_optimizers(self):
         return torch.optim.Adam(self.parameters(), lr=self.hparams.learning_rate)
 
     @staticmethod
     def add_model_specific_args(parent_parser):
         parser = ArgumentParser(parents=[parent_parser], add_help=False)
-        parser.add_argument('--hidden_dim', type=int, default=128,
-                            help='itermediate layers dimension before embedding for default encoder/decoder')
-        parser.add_argument('--latent_dim', type=int, default=32,
-                            help='dimension of latent variables z')
-        parser.add_argument('--input_width', type=int, default=28,
-                            help='input image width - 28 for MNIST (must be even)')
-        parser.add_argument('--input_height', type=int, default=28,
-                            help='input image height - 28 for MNIST (must be even)')
-        parser.add_argument('--batch_size', type=int, default=32)
-        parser.add_argument('--num_workers', type=int, default=8, help="num dataloader workers")
-        parser.add_argument('--learning_rate', type=float, default=1e-3)
-        parser.add_argument('--data_dir', type=str, default='')
+        parser.add_argument(
+            "--hidden_dim",
+            type=int,
+            default=128,
+            help="itermediate layers dimension before embedding for default encoder/decoder",
+        )
+        parser.add_argument("--latent_dim", type=int, default=32, help="dimension of latent variables z")
+        parser.add_argument("--learning_rate", type=float, default=1e-3)
         return parser
 
 
-def cli_main():
+def cli_main(args=None):
+    # cli_main()
     parser = ArgumentParser()
+    parser.add_argument("--dataset", default="mnist", type=str, help="mnist, cifar10, stl10, imagenet")
+    script_args, _ = parser.parse_known_args(args)
+
+    if script_args.dataset == "mnist":
+        dm_cls = MNISTDataModule
+    elif script_args.dataset == "cifar10":
+        dm_cls = CIFAR10DataModule
+    elif script_args.dataset == "stl10":
+        dm_cls = STL10DataModule
+    elif script_args.dataset == "imagenet":
+        dm_cls = ImagenetDataModule
+
+    parser = dm_cls.add_argparse_args(parser)
     parser = Trainer.add_argparse_args(parser)
     parser = AE.add_model_specific_args(parser)
-    args = parser.parse_args()
+    args = parser.parse_args(args)
 
-    ae = AE(**vars(args))
+    dm = dm_cls.from_argparse_args(args)
+    model = AE(*dm.size(), **vars(args))
     trainer = Trainer.from_argparse_args(args)
-    trainer.fit(ae)
+    trainer.fit(model, dm)
+    return dm, model, trainer
 
 
-if __name__ == '__main__':
-    cli_main()
+if __name__ == "__main__":
+    dm, model, trainer = cli_main()
```

Comment:
do we need output here?
```suggestion
    cli_main()
```